common/kernel/nextpnr_types.h

```diff
@@ -180,6 +191,20 @@ enum PortType
     PORT_INOUT = 2
 };
 
+[[maybe_unused]] static const std::string portType_to_str(PortType typ)
+{
+    switch (typ) {
+    case PORT_IN:
+        return "PORT_IN";
+    case PORT_OUT:
+        return "PORT_OUT";
+    case PORT_INOUT:
+        return "PORT_INOUT";
+    default:
+        log_error("Impossible PortType");
```

Comment:
These functions are fine, but `NPNR_ASSERT_FALSE("Impossible PortType");` would be more idiomatic nextpnr code than `log_error`, also to avoid bringing `log.h` into here.